File: examples/elixir/user-server/lib/user_service_impl.ex
defmodule UserServiceImpl do
  @behaviour UserServiceBehaviour

  alias UserServiceTypes

  @impl true
  def handle_create_user(_ctx, _input, _meta), do: {:error, :not_implemented}

  @impl true
  def handle_delete_user(_ctx, _input, _meta), do: {:error, :not_implemented}

  @impl true
  def handle_get_user(_ctx, _input, _meta), do: {:error, :not_implemented}

  @impl true
  def handle_list_users(_ctx, _input, _meta), do: {:error, :not_implemented}

  @impl true
  def handle_update_user(_ctx, _input, _meta), do: {:error, :not_implemented}
end

File: examples/erlang/user-server/src/user_service_impl.erl
-module(user_service_impl).
-behaviour(user_service_behaviour).

-include("user_service_types.hrl").

-export([
    handle_create_user/3,
    handle_delete_user/3,
    handle_get_user/3,
    handle_list_users/3,
    handle_update_user/3
]).

handle_create_user(_Ctx, _Input, _Meta) ->
    {error, not_implemented}.

handle_delete_user(_Ctx, _Input, _Meta) ->
    {error, not_implemented}.

handle_get_user(_Ctx, _Input, _Meta) ->
    {error, not_implemented}.

handle_list_users(_Ctx, _Input, _Meta) ->
    {error, not_implemented}.

handle_update_user(_Ctx, _Input, _Meta) ->
    {error, not_implemented}.
